llama_hub/tools/metaphor/base.py

```diff
@@ -0,0 +1,100 @@
+"""Metaphor tool spec."""
+
+from llama_index.tools.tool_spec.base import BaseToolSpec
+from typing import Optional, List
+from llama_index.readers.schema.base import Document
+import requests
+import urllib.parse
+import datetime
+
+class MetaphorToolSpec(BaseToolSpec):
+    """Metaphor tool spec."""
+
+    spec_functions = ["metaphor_search", "retrieve_documents", "current_date", "find_similar"]
+
+    def __init__(self, api_key: str) -> None:
+        """Initialize with parameters."""
+        from metaphor_python import Metaphor
+        self.client = Metaphor(api_key=api_key)
+
+    def metaphor_search(
+        self,
+        query: str,
```

Comment:
The other useful field that I might put here is `type`, although that will complicate how the LLM perceives the tool. `type` can be `keyword` or `neural`, giving ultimate flexibility as to how to call Metaphor. But I guess there's a tradeoff with ease of use of the tool. "Keyword" search could also be its own separate tool. Maybe named `traditional_search` or something like that.

Thoughts?

Reply:
I'll play around with this a bit tomorrow and let you know what seems most effective

Reply:
If we enable useAutoprompt, does the `type` field still have an effect?

Reply:
if `type` is `keyword` and autoprompt is on, autoprompt won't do anything. I wonder if a better thing to do would be to convert the input into a google-style query, but I imagine the LLM will already have done that

Reply:
@jeffzwang did you imagine that type="keyword" or type="neural" would be set by the LLM or by the human? 

if set by the LLM, we would need to prompt it on best practices onwhen to use keyword or neural. 

otherwise we have two options from my understanding:
- use autoprompt=True
- let user manually set type in `__init__`

out of curiosity is the reason you don't like autoprompt=True because it's slow? that would be a fair consideation

Reply:
It can be set by either the LLM or the human - if set by the LLM, it requires a ~150 token prompt. So not sure what the tradeoff is. We can handle it with a special search type that we have that can auto-decide between the two, with not _that_ much latency overhead.

But yes, autoprompt in general adds a good amount of latency - brings our search from the 250-300ms range to the 600-1000ms range. What do you think is the right tradeoff for LlamaIndex users?